data_structures/src/serializers.rs

```diff
@@ -0,0 +1,572 @@
+extern crate flatbuffers;
+
+use crate::flatbuffers::protocol_generated::protocol::{
+    get_root_as_message, Address as FlatBufferAddress, AddressArgs, Command as FlatBuffersCommand,
+    GetPeers as FlatBuffersGetPeers, GetPeersArgs, IpAddress as FlatBuffersIpAddress,
+    Ipv4 as FlatBuffersIpv4, Ipv4Args as FlatBuffersIpv4Args, Ipv6 as FlatBuffersIpv6,
+    Ipv6Args as FlatBuffersIpv6Args, Message as FlatBuffersMessage, MessageArgs,
+    Peers as FlatBuffersPeers, Peers, PeersArgs, Ping as FlatBuffersPing, PingArgs,
+    Pong as FlatBuffersPong, PongArgs, Verack as FlatBuffersVerack, VerackArgs,
+    Version as FlatBuffersVersion, VersionArgs,
+};
+use crate::types::{
+    Address, Command,
+    IpAddress::{Ipv4 as WitnetIpv4, Ipv6 as WitnetIpv6},
+    Message,
+};
+
+use flatbuffers::FlatBufferBuilder;
+
+const FTB_SIZE: usize = 1024;
+
+#[derive(Debug, Clone, Copy)]
+struct GetPeersFlatbufferArgs {
+    magic: u16,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct GetPeersWitnetArgs {
+    magic: u16,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct PeersFlatbufferArgs<'a> {
+    magic: u16,
+    peers: &'a [Address],
+}
+
+#[derive(Debug, Clone, Copy)]
+struct PeersWitnetArgs<'a> {
+    magic: u16,
+    peers: Peers<'a>,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct PingFlatbufferArgs {
+    magic: u16,
+    nonce: u64,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct PingWitnetArgs {
+    nonce: u64,
+    magic: u16,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct PongFlatbufferArgs {
+    magic: u16,
+    nonce: u64,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct PongWitnetArgs {
+    nonce: u64,
+    magic: u16,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct VerackFlatbufferArgs {
+    magic: u16,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct VerackWitnetArgs {
+    magic: u16,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct VersionFlatbufferArgs<'a> {
+    capabilities: u64,
+    genesis: u64,
+    last_epoch: u32,
+    magic: u16,
+    nonce: u64,
+    receiver_address: &'a Address,
+    sender_address: &'a Address,
+    timestamp: u64,
+    user_agent: &'a str,
+    version: u32,
+}
+
+#[derive(Debug, Clone)]
+struct VersionWitnetArgs {
+    capabilities: u64,
+    genesis: u64,
+    last_epoch: u32,
+    magic: u16,
+    nonce: u64,
+    receiver_address: Address,
+    sender_address: Address,
+    timestamp: u64,
+    user_agent: String,
+    version: u32,
+}
+
+pub trait FromBytes<T> {
+    fn from_bytes(bytes: Vec<u8>) -> Option<T>;
+}
+
+pub trait ToBytes {
+    fn to_bytes(&self) -> Vec<u8>;
+}
+
+// #[cfg(feature = "flatbuffers")]
+impl FromBytes<Message> for Message {
+    fn from_bytes(bytes: Vec<u8>) -> Option<Self> {
```

Comment:
I would suggest using a slice as parameter (unless we want the vector to be moved in here, which might be our case). This way both vectors and array slices could be used as parameters to this function.

```suggestion
    fn from_bytes(bytes: &[u8]) -> Option<Self> {
```